### PR TITLE
feat(swagger): implement SendError for BaseAPI handler

### PR DESCRIPTION
**Pull request by heww**
implement SendError for BaseAPI handler

Signed-off-by: He Weiwei <hweiwei@vmware.com>

### CODE REVIEW

File: src/server/error/error.go [hunk 1]
```diff
@@ -51,3 +47,33 @@ func getHTTPStatusCode(errCode string) int {
...
+	rw.WriteHeader(getHTTPStatusCode(code))
+
+	var e *ierror.Error
+	if !errors.As(r.err, &e) {
+		e = &ierror.Error{
+			Code:    code,
+			Message: r.err.Error(),
+		}
+	}
+
```
Comment (ywk253100) [inline src/server/error/error.go]: Call `APIError()` instead?

File: src/server/error/error.go [hunk 2]
```diff
@@ -51,3 +47,33 @@ func getHTTPStatusCode(errCode string) int {
...
+}
+
+// WriteResponse ...
+func (r *ErrResponder) WriteResponse(rw http.ResponseWriter, producer runtime.Producer) {
+	code := ierror.ErrCode(r.err)
+	rw.WriteHeader(getHTTPStatusCode(code))
+
+	var e *ierror.Error
+	if !errors.As(r.err, &e) {
+		e = &ierror.Error{
```
Comment (ywk253100) [inline src/server/error/error.go]: The response should be an error array, is this covered by other logic?

File: src/server/v2.0/handler/base.go [hunk 3]
```diff
@@ -30,6 +31,5 @@ func (*BaseAPI) Prepare(ctx context.Context, operation string, params interface{
 
 // SendError returns response for the err
 func (*BaseAPI) SendError(ctx context.Context, err error) middleware.Responder {
-	// TODO: implement send error
-	return nil
+	return errs.NewErrResponder(err)
```
Comment (ywk253100) [inline src/server/v2.0/handler/base.go]: Let's move this file out of v2.0 as it is shared by all versions of APIs

File: src/server/error/error.go (hunk 2 above)
Comment (heww) [reply src/server/error/error.go]: No, management APIs (v2.0) not needed keep the same as the oci error response.

File: src/server/v2.0/handler/base.go (hunk 3 above)
Comment (heww) [reply src/server/v2.0/handler/base.go]: Yes, it is shared by all versions of APIs (swagger generated APIs) so there is a `ErrResponder` in `server/error` pkg, `SendError` in `BaseAPI` is a shortcut.

File: src/server/error/error.go (hunk 2 above)
Comment (ywk253100) [reply src/server/error/error.go]: I think @wy65701436 has already implemented to return the error array for Harbor APIs in PR https://github.com/goharbor/harbor/pull/10324?

I'm OK for both these two solutions, but we must be consistent.

**Pull request merged**